test/cmdLineTests/shareClassTests/SCCMLTests/SharedClassesModularityTests.xml

```diff
@@ -0,0 +1,148 @@
+		<output type="success" caseSensitive="yes" regex="no">has been destroyed</output>
+		<output type="success" caseSensitive="yes" regex="no">is destroyed</output>
+		<output type="failure" caseSensitive="no" regex="no">Unhandled Exception</output>
+		<output type="failure" caseSensitive="yes" regex="no">Exception:</output>
+		<output type="failure" caseSensitive="no" regex="no">corrupt</output>
+		<output type="failure" caseSensitive="yes" regex="no">Processing dump event</output>
+	</test>
+	
+	<test id="Test 1 set-up : remove one class from patch path" timeout="600" runPath=".">
+		<exec command="rm -f $DISK_CLASS$" />
+		<command>ls $HANOI_DIR$</command>
+		<output type="success" caseSensitive="yes" regex="no">Post.class</output>
+		<output type="failure" caseSensitive="no" regex="no">Disk.class</output>
+	</test>
+
+	<test id="Test 1: Test --patch-module for Bootstrap ClassLoader and Builtin ClassLoader. Patched classes should not be stored to the cache. Other classes from the patched module should still be stored" timeout="600" runPath=".">
+			<!-- module java.base and module utils are patched -->
+		<command>$JAVA_EXE$ $currentMode$,verboseIO -verbose --patch-module java.base=$JAVALIB_DIR$$PATHSEP$jrt-fs.jar --patch-module utils=. --module-path $UTILSJAR$ -m utils/$PROGRAM_HANOI$</command>
+		<output type="success" caseSensitive="yes" regex="no">Puzzle solved!</output>
+			<!-- Bootstrap ClassLoader: classes in java.base that are not patched (loaded from Jimage) should be stored to the cache -->
+		<output type="required" caseSensitive="yes" regex="yes" javaUtilPattern="yes">Stored class java/.* in shared cache for class-loader id 0 with URL .*[\\/]lib[\\/]modules</output>
+			<!-- Builtin ClassLoader: classes in utils that are not patched (loaded from module path) should be stored to the cache (in this case class Disk) -->
+		<output type="required" caseSensitive="yes" regex="yes" javaUtilPattern="yes">Stored class org/openj9/test/ivj/Disk in shared cache for class-loader id [2-9] with URL .*[\\/]utils.jar</output>
+			<!-- Make sure patched classes in module java.base are loaded by bootstrap ClassLoader -->
+		<output type="required" caseSensitive="yes" regex="yes" javaUtilPattern="yes">class load: jdk/internal/jimage.* from: .*[\\/]jrt-fs.jar</output>
+			<!-- make sure patched class (class Post) in module utils is loaded by Builtin ClassLoader -->
+		<output type="required" caseSensitive="yes" regex="no">class load: org.openj9.test.ivj.Post from:</output>
+
+			<!-- Bootstrap ClassLoader: patch classes in java.base should not be stored to the cache -->
+		<output type="failure" caseSensitive="no" regex="yes" javaUtilPattern="yes">Stored class jdk/internal/jimage.*[\\/]jrt-fs.jar</output>
+			<!-- Builtin ClassLoader: patched classes in module utils should not be stored to the cache -->
+		<output type="failure" caseSensitive="no" regex="yes" javaUtilPattern="yes">Stored class .*/ivj/Post in shared cache .*utils.jar</output>
```

Comment:
A failure condition doesn't need to be so specific, this could be minimized to `Stored class .*/ivj/Post in shared cache`